### PR TITLE
Fix various types in TypeScript declaration file...

### PR DESCRIPTION
**Pull request by anfriis**
Refactor MapView, CameraProps, UserLocation, and other minor fixes

### CODE REVIEW

**Comment by anfriis** [comment]
I see that you have a index.d.js generator setup, but until it is working I think this should be merged in so the framework is actually usable in projects with TypeScript. I have used this type file in my own implementation to test it

**Comment by mfazekas** [comment]
@kristfal, @mattijsf do you use typescript? I think it would be good to delegate someone from community who actively uses typescript, to manage typescript related PR-s.

I'd prefer automatic typescript generation from doc comments, but since it's not there yet, i'm not against merging PR with improvements to ts files. But it would be good if it was done someone who knows better than me what's going on exactly.

**Comment by kristfal** [comment]
I’m not running TS, so no strong opinion either way.

**Comment by mattijsf** [comment]
In general maintaining these .ts declaration files by hand doesn't sound great indeed.

I don't have experience with TypeScript. I do have enough experience with Flow to know that I might give TypeScript a try in future (new) projects. But until then I don't have enough experience to provide help.

**Comment by henninghall** [comment]
It would be great if this PR could be merged. The current typescript declaration included in the project is quite outdated. 

[Here is a related PR]( https://github.com/react-native-mapbox-gl/maps/pull/145) which probably could be closed if this one (230) is merged. This PR seems a bit more complete to me since it also removes outdated props in MapViewProps.

File: index.d.ts [hunk 1]
```diff
@@ -172,6 +199,24 @@ interface MapViewProps extends ViewProperties {
     onUserTrackingModeChange?: () => void;
 }
 
+interface CameraProps {
+    animationDuration?: number;
+    animationMode?: string;
```
Comment (henninghall) [inline index.d.ts]: Could be improved by only allowing valid strings: 
`animationMode?: 'flyTo' | 'easeTo' | 'moveTo';`

File: index.d.ts [hunk 2]
```diff
@@ -172,6 +199,24 @@ interface MapViewProps extends ViewProperties {
     onUserTrackingModeChange?: () => void;
 }
 
+interface CameraProps {
```
Comment (henninghall) [inline index.d.ts]: I guess this should extend ViewProps? see [Camera.js#L69](https://github.com/react-native-mapbox-gl/maps/blob/master/javascript/components/Camera.js#L69)

File: index.d.ts [hunk 3]
```diff
@@ -172,6 +199,24 @@ interface MapViewProps extends ViewProperties {
...
+    animationDuration?: number;
+    animationMode?: string;
+    centerCoordinate?: Array<number>;
+    heading?: number;
+    pitch?: number;
+    zoomLevel?: number;
+    minZoomLevel?: number;
+    maxZoomLevel?: number;
+}
```
Comment (henninghall) [inline index.d.ts]: It seems like some props related to user tracking are missing, they might be added after this PR, I don't know. See [Camera.js from line 85](https://github.com/react-native-mapbox-gl/maps/blob/master/javascript/components/Camera.js#L85)

**Comment by mfazekas** [comment]
@henninghall would you be interested in handling type script related PR-s? If so i can add you to the project.

**Comment by henninghall** [comment]
Sure, I can handle it for a while. Can't promise any long term commitment though 🙂

**Comment by anfriis** [comment]
Thanks for the suggested changes @henninghall! 
I have pushed a new commit with the changes. I unfortunately didn't have time to change and test all types in the file, so feel free to pick up on that 👍

**Comment by mfazekas** [comment]
@henninghall i've invited to the repo, feel free to handle TS related PR's.

**Comment by henninghall** [comment]
> Thanks for the suggested changes @henninghall!
> I have pushed a new commit with the changes. I unfortunately didn't have time to change and test all types in the file, so feel free to pick up on that 👍

Great, I will test it out!




> @henninghall i've invited to the repo, feel free to handle TS related PR's.

Thanks, I will do that! Just tag me if you realize that a particular PR needs typescript updates that I might have missed.

**Comment by henninghall** [comment]
Thanks @anfriis! We still have some typescript left to write to cover the entire library. For instance `ImageSource` is still missing.

**Pull request merged**